internal/host/hostcommands/install_cmd.go

```diff
@@ -276,8 +274,38 @@ func constructHostInstallerArgs(cluster *common.Cluster, host *models.Host, log
 	return toJSONString(installerArgs)
 }
 
+func (i *installCmd) appendDHCPArgs(cluster *common.Cluster, installerArgs []string) ([]string, error) {
+	if hasUserConfiguredIP(installerArgs) {
+		return installerArgs, nil
+	}
+	if swag.BoolValue(cluster.UserManagedNetworking) && cluster.Hosts == nil {
+		if err := i.loadHosts(cluster); err != nil {
+			return installerArgs, err
+		}
+	}
+	machineNetworkCIDR := network.GetMachineCidrForUserManagedNetwork(cluster, i.log)
+	i.log.Debugf("Machine network CIDR: %s. IPv6: %t", machineNetworkCIDR, network.IsIPv6CIDR(machineNetworkCIDR))
+	if network.IsIPv6CIDR(machineNetworkCIDR) {
+		return append(installerArgs, "--append-karg", "ip=dhcp6"), nil
+	} else if machineNetworkCIDR != "" {
+		return append(installerArgs, "--append-karg", "ip=dhcp"), nil
+	}
+	return installerArgs, errors.Errorf("cannot determine machine network address family")
+}
+
+func (i *installCmd) loadHosts(cluster *common.Cluster) error {
+
+	var hosts []*models.Host
+	if err := i.db.Find(&hosts, "cluster_id = ?", cluster.ID).Error; err != nil {
```

Comment:
we have  a number of places in the code where we need to load hosts. Maybe we can move this code to the common?

Reply:
i think we have it already

Reply:
I don't think it's worth it, because it's just two lines, and the error handling may differ.

Reply:
@tsorya no we don't, there're three places where we fetch hosts this way, including this one. One of them handles the error a bit differently. IMO the API of `Find()` is concise enough so that we'll gain nothing by wrapping it in another function. On the contrary, having a function that runs the query, writes to the log and returns the error may potentially lead to double-logging. Yet without writing to the log it will be reduced to just 
```var hosts []*models.Host
   return db.Find(&hosts, "cluster_id = ?", clusterID)```

Reply:
and why we can't load them with cluster?

Reply:
i mean, if we already bring cluster , let preload hosts there

Reply:
i think we can use GetClusterFromDBWithoutDisabledHosts when we get cluster and it will be good enough.